packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -0,0 +1,172 @@
+    this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
+    this.handlePrimaryKeyDown = this.handlePrimaryKeyDown.bind(this);
+    this.handlePrimaryKeyUp = this.handlePrimaryKeyUp.bind(this);
+    this.handleChevronKeyDown = this.handleChevronKeyDown.bind(this);
+    this.handleChevronKeyUp = this.handleChevronKeyUp.bind(this);
+
+    this.state = { isOpen: false, chevronActive: false, primaryActive: false };
+  }
+
+  handleDropdownButtonClick() {
+    this.setState({ isOpen: true });
+  }
+
+  handleDropdownRequestClose() {
+    this.setState({ isOpen: false });
+  }
+
+  /*
+    In FireFox active styles don't get applied onKeyDown
+   */
+  handlePrimaryKeyDown(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      this.setState({ primaryActive: true });
+    }
+  }
+
+  handlePrimaryKeyUp(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      this.setState({ primaryActive: false });
+    }
+  }
+
+  handleChevronKeyDown(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE) {
+      this.setState({ chevronActive: true });
+    }
+  }
+
+  handleChevronKeyUp(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE) {
+      this.setState({ chevronActive: false });
+    }
+  }
+
+  render() {
+    const {
+      children,
+      disabled,
+      isBlock,
+      primaryOptionLabel,
+      onClick,
+      variant,
+      ...customProps
+    } = this.props;
+
+    const {
+      isOpen,
+      primaryActive,
+      chevronActive,
+    } = this.state;
+
+    const primaryClassnames = cx(
+      'split-button-primary',
+      variant,
+      { 'is-block': isBlock },
+      { 'is-active': primaryActive },
+    );
+    const chevronClassnames = cx(
+      'split-button-chevron',
+      variant,
+      { 'is-active': isOpen || chevronActive },
+    );
+
+    return (
+      <DropdownButtonBase
+        {...customProps}
+        buttons={children}
+        isOpen={isOpen}
+        requestClose={this.handleDropdownRequestClose}
+        disabled={disabled}
+        isBlock={isBlock}
+      >
+        <button
+          type="button"
+          className={primaryClassnames}
+          onClick={onClick}
+          onKeyDown={this.handlePrimaryKeyDown}
+          onKeyUp={this.handlePrimaryKeyUp}
+          disabled={disabled}
+          tabIndex={disabled ? '-1' : undefined}
+          aria-disabled={disabled}
+        >
+          {primaryOptionLabel}
+        </button>
+        <button
+          type="button"
+          onClick={this.handleDropdownButtonClick}
+          onKeyDown={this.handleChevronKeyDown}
+          onKeyUp={this.handleChevronKeyUp}
+          className={chevronClassnames}
+          disabled={disabled}
+          tabIndex={disabled ? '-1' : undefined}
+          aria-disabled={disabled}
+          aria-expanded={isOpen || undefined}
+          aria-haspopup
+          aria-label="More Options"
```

Comment:
We'll want to get translations for this.

Reply:
Submitted a translations request